zulip_bots/zulip_bots/bots/calendarbot/googlecalendar.py

```diff
@@ -1,61 +1,107 @@
 from __future__ import print_function
 
-import datetime
+import logging
 import os.path
 from dataclasses import dataclass
 from datetime import datetime, time
 
 from google.auth.transport.requests import Request
-from google.oauth2 import service_account
 from google.oauth2.credentials import Credentials
 from google_auth_oauthlib.flow import InstalledAppFlow
 from googleapiclient.discovery import build
 from googleapiclient.errors import HttpError
+from dataclasses import dataclass
+
+logging.basicConfig(
+    filename="zulip_bots/zulip_bots/bots/calendarbot/calendarbot.log",
+    encoding="utf-8",
+    level=logging.DEBUG,
+)
 
 # If modifying these scopes, delete the file token.json.
 SCOPES = ["https://www.googleapis.com/auth/calendar"]
 
 from google.oauth2 import service_account
 
 SCOPES = ["https://www.googleapis.com/auth/calendar"]
-SERVICE_ACCOUNT_FILE = "python-zulip-api/zulip_bots/zulip_bots/bots/calendarbot/creds.json"
+SERVICE_ACCOUNT_FILE = "zulip_bots/zulip_bots/bots/calendarbot/creds.json"
+BOT_CALENDAR_ID = (
+    "5edeaa7e7808543c6f5b1f64433d135e618cc3cad5d2c2f2df2b452c81957459@group.calendar.google.com"
+)
 
 
-def send_google_invite(meeting_details):
-    creds = None
-    # The file token.json stores the user's access and refresh tokens, and is
-    # created automatically when the authorization flow completes for the first
-    # time.
-    if os.path.exists("token.json"):
-        creds = Credentials.from_authorized_user_file("token.json", SCOPES)
-    # If there are no (valid) credentials available, let the user log in.
-    if not creds or not creds.valid:
-        if creds and creds.expired and creds.refresh_token:
-            creds.refresh(Request())
-        else:
-            flow = InstalledAppFlow.from_client_secrets_file(SERVICE_ACCOUNT_FILE, SCOPES)
-            creds = flow.run_local_server(port=8080)
-        # Save the credentials for the next run
-        with open("token.json", "w") as token:
-            token.write(creds.to_json())
+def authenticate_google():
+    # TODO we need to prevent the authenticate refresh flow from running for end users
 
     try:
-        calendar = build("calendar", "v3", credentials=creds)
-        print(meeting_details.invitees)
+        global creds
+        creds = None
+        # The file token.json stores the user's access and refresh tokens, and is
+        # created automatically when the authorization flow completes for the first
+        # time.
+        if os.path.exists("token.json"):
+            creds = Credentials.from_authorized_user_file("token.json", SCOPES)
```

Comment:
Just curious - does this not raise a `ValueError` if it cannot process the `token.json` into valid constructed credentials? I'm sure you've been able to trigger the next block, so I guess it doesn't?